rewrite-javascript/rewrite/src/javascript/parser.ts

```diff
@@ -55,15 +57,17 @@ import {
 import {JavaScriptTypeMapping} from "./typeMapping";
 import {produce} from "immer";
 import Kind = JS.Kind;
+import {RpcCodecs} from "../rpc";
 
 export class JavaScriptParser extends Parser {
 
     private readonly compilerOptions: ts.CompilerOptions;
     private readonly sourceFileCache: Map<string, ts.SourceFile> = new Map();
     private oldProgram?: ts.Program;
 
-    constructor() {
-        super();
+    constructor(ctx: ExecutionContext = new ExecutionContext(),
+                readonly relativeTo?: string) {
```

Comment:
I don't really like this part. I think it would probably be better if the `Parser` class defined the constructor as something like:

<img width="470" alt="image" src="https://github.com/user-attachments/assets/dc190312-64d0-431b-9677-d8186465b432" />

I think that should allow the `JavaScriptParser` to extend that `ParserOptions` type for its own options.